test/e2e/framework/pod/create.go

```diff
@@ -50,13 +50,23 @@ func CreateWaitAndDeletePod(c clientset.Interface, ns string, pvc *v1.Persistent
 		}
 	}()
 
-	err = TestPodSuccessOrFail(c, ns, runPod)
+	err = testPodSuccessOrFail(c, ns, runPod)
 	if err != nil {
 		return fmt.Errorf("pod %q did not exit with Success: %v", runPod.Name, err)
 	}
 	return // note: named return value
 }
 
+// testPodSuccessOrFail tests whether the pod's exit code is zero.
+func testPodSuccessOrFail(c clientset.Interface, ns string, pod *v1.Pod) error {
+	e2elog.Logf("Pod should terminate with exitcode 0 (success)")
+	if err := WaitForPodSuccessInNamespace(c, pod.Name, ns); err != nil {
+		return fmt.Errorf("pod %q failed to reach Success: %v", pod.Name, err)
+	}
+	e2elog.Logf("Pod %v succeeded ", pod.Name)
+	return nil
+}
+
```

Comment:
Cleaning up files is a good thing but I think it would be better to organize functions in a way that makes them easier to discover. I.e., functions that create pods in create.go, functions that test for conditions in test*.go, etc.

AT this moment we may only have the one file for testing/verifying but more things may find a home there later on.
/hold

Reply:
Additional comments  https://github.com/kubernetes/kubernetes/pull/83814#discussion_r334217997